File: dcp-solutions/2025-01-03/solution.js
```javascript
class TreeNode {
  constructor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

function isSubtree(s, t) {
  const sStr = getTreeToArray(s);
  const tStr = getTreeToArray(t);

  return sStr.includes(tStr);
}

function getTreeToArray(root) {
  if (!root) return '';

  const left = getTreeToArray(root.left);
  const right = getTreeToArray(root.right);

  return (left + '' + root.val + '' + right).split('').sort().join('');
}
```

File: dcp-solutions/2025-01-03/solution.rb
```ruby
class TreeNode
  attr_accessor :val, :left, :right

  def initialize(val = 0, left = nil, right = nil)
    @val = val
    @left = left
    @right = right
  end
end

def isSubtree(s, t)
  return true if s == nil
  return false if t == nil

  return isMatch(s, t) ? true : false
end

def isMatch(t1, t2)
  return false unless t1 && t2
  return t1.val == t2.val &&
         (isMatch(t1.left,  t2.left) || isMatch(t1.right, t2.right))
end
```